src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -257,11 +289,37 @@ def create_key(self, arg0, arg1=None, arg2=None, names='i,j,k'):
             a = K(v[0])
             b = K(v[1])
 
-        # QuaternionAlgebra(K, a, b)
         else:
+            # QuaternionAlgebra(K, primes, inv_archimedean)
             K = arg0
-            a = K(arg1)
-            b = K(arg2)
+            if category(K) is not NumberFields():
+                raise ValueError("quaternion algbera must be defined over a number field")
```

Comment:
```
sage: category(QQ)
Join of Category of number fields and Category of quotient fields and Category of metric spaces
```

Is it possible to use this to check if K is QQ or a number field?

Reply:
It might be possible, but even then I would advise against it: You need to handle the case of the rational field separately anyways (I ran into the same issue in #37173), and up to the check `is_RationalField(K)` I don't see any code that doesn't work for either setting; hence it is probably the easiest approach to place the above category check in the else block after the treatment of the rational field case.